notebooks/stac-load-S2-deafrica.py

```diff
@@ -0,0 +1,154 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.6
+#   kernelspec:
+#     display_name: 'Python 3.8.12 64-bit (''stac'': conda)'
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Access Sentinel 2 Analysis Ready Data from Digital Earth Africa
+#
+# [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/opendatacube/odc-stac/develop?labpath=notebooks%2Fstac-load-S2-deafrica.ipynb)
+#
+# https://explorer.digitalearth.africa/products/s2_l2a
+
+# %% [markdown]
+# ## Import required packages
+
+# %%
+import rasterio
+from pystac_client import Client
+
+from odc.stac import stac_load
+
+# %% [markdown]
+# ## Set configuration
+#
+# The configuration dictionary is determined from the product's definition, availble at https://explorer.digitalearth.africa/products/s2_l2a#definition-doc
+#
+# All assets except SLC have the same configuration. SLC uses `uint8` rather than `uint16`.
+#
+# In the configuration, we also supply the aliases for each band. This means we can load data by band name rather than band number.
+
+# %%
+config = {
+    "s2_l2a": {
+        "assets": {
+            "*": {
+                "data_type": "uint16",
+                "nodata": 0, 
+                "unit": "1",
+            },
+            "SLC": {
+                "data_type": "uint8",
+                "nodata": 0, 
+                "unit": "1",
+            },
+        },
+        "aliases": {
+            "costal_aerosol": "B01",
+            "blue": "B02",
+            "green": "B03",
+            "red": "B04",
+            "red_edge_1": "B05",
+            "red_edge_2": "B06",
+            "red_edge_3": "B07",
+            "nir": "B08",
+            "nir_narrow": "B08A",
+            "water_vapour": "B09",
+            "swir_1": "B11",
+            "swir_2": "B12",
+            "mask": "SLC",
+            "aerosol_optical_thickness": "AOT",
+            "scene_average_water_vapour": "WVP",
+        }
+    }
+}
+
+# %% [markdown]
+# ## Connect to the Digital Earth Africa STAC catalog
+
+# %%
+# Open the stac catalogue
+deafrica_stac_address = 'https://explorer.digitalearth.africa/stac'
+catalog = Client.open(deafrica_stac_address)
```

Comment:
For `deafrica_stac_address` I would either not use a variable at all, just pass in url string into `.open(..)`, or use a more generic name, i.e. without `deafrica_`. Like `stac_api_endpoint = `, if there were several stac end points within this notebook, then this name would be fine, but since it's only one having an extra specifier is not helpful. Having more generic names helps with cut and pasting as well.

Reply:
Done